lib/order.rb

```diff
@@ -0,0 +1,33 @@
+class Order
+
+	attr_reader :my_order
+
+	def initialize
+		@my_order = {}
+	end
+
+	def select(restaurant, dish)
+		restaurant.dishes.each{
+			|key, value|
+			if key == dish
+				if @my_order.has_key? key
+					nuVal = (@my_order.fetch(key)) + value
+					@my_order.store(key,nuVal)
+				else
+					@my_order.store(key,value)
+				end
+			end
+		}
+	end
```

Comment:
I'm not sure, my though process would be select an item from the menu, then shovel << to the my_order hash. Maybe then you wouldn't have to use .each?

Reply:
I don't think you can use << on a hash, I'm pretty sure that's just for arrays. Why would you advise against using .each?